dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/event/data/JdbcEventAnalyticsManager.java

```diff
@@ -628,4 +672,26 @@ private String resolveCoordinateFieldColumnName( String coordinateField, EventQu
         }
         return coordinateField;
     }
+
+    protected static class ExceptionHandler
```

Comment:
Could be this Exception Handler be moved at the `CrudControllerAdvice` level? The `CrudControllerAdvice` is annotated with `@ControllerAdvice` and it is applied to all the controllers in the system.
I'm just thinking here that perhaps this handler is very specific to ANalytics, so it's better to keep it separated from the global exception handling strategy.
In any case, I would perhaps make the `ExceptionHandler` class its own class, which could be perhaps applied to all *AnalyticsManager in the system.

Reply:
Hey @luciano-fiandesio,
this method is just deciding which exception should be thrown. It's very specific to this situation.
Once the exception is thrown, it will be handled by the CrudControllerAdvice:

`@ExceptionHandler( QueryRuntimeException.class )
    public void queryRuntimeExceptionHandler( QueryRuntimeException ex, HttpServletResponse response, HttpServletRequest request )
    {
        webMessageService.send( WebMessageUtils.conflict( ex.getMessage(), ex.getErrorCode() ), response, request );
    }`

Reply:
Regarding the second point, this is could be done. But would require a great amount of refactoring. I would say we can create a task for it if you guys agree, and we can tackle as part of another PR.
What do you think?

Reply:
This sounds like a task for a different issue/PR. I will merge this PR. @maikelarabori you can create a new issue for the second part of that comment and discuss further with Luciano :)